distributed/dashboard/scheduler.py

```diff
@@ -268,8 +270,9 @@ def collect(self):
             "Number of tasks known by scheduler.",
             labels=["state"],
         )
-        tasks.add_metric(["received"], len(self.server.tasks))
-        tasks.add_metric(["unrunnable"], len(self.server.unrunnable))
+        task_counter = Counter(map(lambda x: x.state, self.server.tasks.values()))
```

Comment:
If we do do it this way (rather than the PR mentioned below) then we might want to use `(cy)toolz.frequencies`, which is a bit faster.

Reply:
@fjetter just to double check: Is that really a counter in the Prometheus sense (i.e. only goes up, ever goes down unless it rests to 0 due to integer overflow or server restart)? 

Given the way I understand Dask I would assume it is only a gauge (i.e. can go up or go down)

Reply:
>  just to double check: Is that really a counter in the Prometheus sense 

It's a `GaugeMetric`. The `Counter` is the python collections container to calculate the frequencies of the individual states.

What's _missing_ here is that we probably want to expose all existing states instead of only those currently existing, i.e. we should ensure there is a value for each state in `ALL_TASK_STATES` which makes the prometheus metric more consistent.

Reply:
This becomes pretty easy (and much cheaper) once we merge in https://github.com/dask/distributed/pull/3262/